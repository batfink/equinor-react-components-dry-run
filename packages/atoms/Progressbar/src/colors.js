import { hsl } from 'polished'

/* eslint-disable no-multi-spaces */
/* eslint-disable key-spacing */

export const colors = {
    white:    { hue: 100, saturation: 1.00, lightness:  1.00 },
    black:    { hue:   0, saturation: 0.00, lightness:  0.20 },
    energy:   { hue: 348, saturation: 1.00, lightness:  0.54 },
    heritage: { hue: 343, saturation: 1.00, lightness:  0.24 },
    slate:    { hue: 206, saturation: 0.32, lightness:  0.21 },
    moss:     { hue: 184, saturation: 1.00, lightness:  0.24 },
    spruce:   { hue:  25, saturation: 1.00, lightness:  0.92 },
    lichen:   { hue: 138, saturation: 0.67, lightness:  0.94 },
    mist:     { hue: 199, saturation: 0.58, lightness:  0.90 },
}

export const white     = hsl(colors.white)
export const black     = hsl(colors.black)
export const energy    = hsl(colors.energy)
export const heritage  = hsl(colors.heritage)
export const slate     = hsl(colors.slate)
export const moss      = hsl(colors.moss)
export const spruce    = hsl(colors.spruce)
export const lichen    = hsl(colors.lichen)
export const mist      = hsl(colors.mist)
