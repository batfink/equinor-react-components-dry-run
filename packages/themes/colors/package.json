{
    "name": "@equinor/colors",
    "main": "dist/colors.js",
    "module": "lib/colors.js",
    "version": "0.0.1",
    "license": "UNLICENSED",
    "scripts": {
        "prepare": "yarn build",
        "clean": "rimraf dist && mkdir dist",
        "bundle": "rollup -c",
        "build": "npm-run-all clean bundle"
    }
}
