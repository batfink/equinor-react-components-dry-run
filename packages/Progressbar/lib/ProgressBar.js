import PropTypes from 'prop-types';
import styled, { keyframes, css } from 'styled-components';
import { tint } from 'polished';
import { slate, mist } from '../../themes/colors';

const tinted = tint(0.7, slate);

const Animation = keyframes`
    0% {
        background-position: 0 0;
    }
    100% {
        background-position: 100% 0;
    }
`;

const animatedBackground = css`
    animation: ${Animation} 6s infinite reverse linear;
    background-image: repeating-linear-gradient(
        -45deg,
        ${tinted},
        ${tinted} 20px,
        ${mist} 20px,
        ${mist} 40px
    );
    background-size: 200% 100%;
`;

const ProgressBar = styled.div`
    background: ${slate};
    border-radius: 5px;
    height: 10px;
    border: 1px solid ${tinted};
    ${({ showAnimation }) => showAnimation && animatedBackground};
`;

ProgressBar.propTypes = { showAnimation: PropTypes.bool };

ProgressBar.defaultProps = { showAnimation: false };

export default ProgressBar;
