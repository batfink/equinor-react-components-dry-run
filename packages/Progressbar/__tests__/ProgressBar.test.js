import React from 'react';
import { slate } from '../../themes/colors';
import ProgressBar from '../lib/ProgressBar';
import renderer from 'react-test-renderer';

describe('ProgressBar', () => {
    it('should render a ProgressBar', () => {
        const tree = renderer.create(<ProgressBar />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render a ProgressBar with animation', () => {
        const tree = renderer.create(<ProgressBar showAnimation />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should have background slate', () => {
        const tree = renderer.create(<ProgressBar />).toJSON();
        expect(tree).toHaveStyleRule('background', slate);
    });

    it('should have background-size 200% 100% when animated', () => {
        const tree = renderer.create(<ProgressBar showAnimation />).toJSON();
        expect(tree).toHaveStyleRule('background-size', '200% 100%');
    });
});
