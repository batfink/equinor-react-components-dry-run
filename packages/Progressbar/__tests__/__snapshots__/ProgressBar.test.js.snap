// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should render a ProgressBar 1`] = `
.c0 {
  background: #243847;
  border-radius: 5px;
  height: 10px;
  border: 1px solid #bdc3c7;
}

<div
  className="c0"
/>
`;

exports[`ProgressBar should render a ProgressBar with animation 1`] = `
.c0 {
  background: #243847;
  border-radius: 5px;
  height: 10px;
  border: 1px solid #bdc3c7;
  -webkit-animation: eirxjL 6s infinite reverse linear;
  animation: eirxjL 6s infinite reverse linear;
  background-image: repeating-linear-gradient( -45deg,#bdc3c7,#bdc3c7 20px,#d7ebf4 20px,#d7ebf4 40px );
  background-size: 200% 100%;
}

<div
  className="c0"
/>
`;
